src/scanpy/plotting/_anndata.py

```diff
@@ -148,69 +149,101 @@ def scatter(
     -------
     If `show==False` a :class:`~matplotlib.axes.Axes` or a list of it.
     """
+    # color can be a obs column name or a matplotlib color specification (or a collection thereof)
+    if color is not None:
+        color = cast(
+            Collection[str | ColorLike],
+            [color] if isinstance(color, str) or is_color_like(color) else color,
+        )
     args = locals()
-    if _check_use_raw(adata, use_raw):
-        var_index = adata.raw.var.index
-    else:
-        var_index = adata.var.index
+
     if basis is not None:
         return _scatter_obs(**args)
     if x is None or y is None:
         raise ValueError("Either provide a `basis` or `x` and `y`.")
-    if (
-        (x in adata.obs.columns or x in var_index)
-        and (y in adata.obs.columns or y in var_index)
-        and (color is None or color in adata.obs.columns or color in var_index)
-    ):
+    if _check_if_annotations(adata, "obs", x=x, y=y, colors=color, use_raw=use_raw):
         return _scatter_obs(**args)
-    if (
-        (x in adata.var.columns or x in adata.obs.index)
-        and (y in adata.var.columns or y in adata.obs.index)
-        and (color is None or color in adata.var.columns or color in adata.obs.index)
-    ):
-        adata_T = adata.T
-        axs = _scatter_obs(
-            adata=adata_T,
-            **{name: val for name, val in args.items() if name != "adata"},
-        )
+    if _check_if_annotations(adata, "var", x=x, y=y, colors=color, use_raw=use_raw):
+        args_t = args.copy()
+        args_t["adata"] = adata.T
```

Comment:
Out of curiosity, why not
```suggestion
        args_t = { **args, "adata": adata.T }
```

Reply:
huh, good point, I thought that’d cause something like the “parameter specified multiple times” error, but it doesn’t.